### PR TITLE
feat(deprecation): fixing deprecation warnings

### PR DESCRIPTION
**Pull request by milos1290**

### CODE REVIEW

**Comment by Enuviel** [comment]
test this please

File: AndroidSDK/src/com/leanplum/internal/SocketIOClient.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
 import java.util.Arrays;
 import java.util.HashSet;
 
+@SuppressWarnings("deprecation")
```
Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/internal/SocketIOClient.java]: could you write some comment why?

File: AndroidSDK/src/com/leanplum/internal/WebSocketClient.java [hunk 2]
```diff
@@ -53,6 +53,7 @@
 import javax.net.ssl.SSLSocketFactory;
 import javax.net.ssl.TrustManager;
 
+@SuppressWarnings("deprecation")
```
Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/internal/WebSocketClient.java]: could you write some comment why?

File: AndroidSDK/src/com/leanplum/LocationManagerImplementation.java [hunk 3]
```diff
@@ -263,7 +264,7 @@ private boolean isMetaDataSet() {
       return false;
     }
   }
-
+  @SuppressWarnings("MissingPermission")
```
Comment (Enuviel) [inline AndroidSDK/src/com/leanplum/LocationManagerImplementation.java]: could you write some comment why?

File: AndroidSDK/src/com/leanplum/internal/Socket.java [hunk 4]
```diff
@@ -48,6 +48,7 @@
  *
  * @author Andrew First, Ben Marten
  */
+@SuppressWarnings("deprecation")
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/Socket.java]: can we please add a comment about why its deprecated?

File: AndroidSDK/src/com/leanplum/internal/ResourceQualifiers.java [hunk 5]
```diff
@@ -94,6 +95,7 @@ public Object getMatch(String str) {
 
       @Override
       public boolean isMatch(Object value, Configuration config, DisplayMetrics display) {
+        //noinspection deprecation
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/ResourceQualifiers.java]: can we please add a comment about why its deprecated/suppressed?

File: AndroidSDK/src/com/leanplum/internal/ResourceQualifiers.java [hunk 6]
```diff
@@ -80,6 +80,7 @@ public Object getMatch(String str) {
...
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/internal/ResourceQualifiers.java]: can we please add a comment about why its deprecated/suppressed?

File: AndroidSDK/src/com/leanplum/LocationManagerImplementation.java (hunk 3 above)
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/LocationManagerImplementation.java]: can we please add a comment about why its suppressed?

File: AndroidSDK/src/com/leanplum/LeanplumResources.java [hunk 7]
```diff
@@ -42,7 +42,7 @@
 
 // Description of resources.asrc file (we don't use this right nwo)
 // http://ekasiswanto.wordpress.com/2012/09/19/descriptions-of-androids-resources-arsc/
-
+@SuppressWarnings("deprecation")
```
Comment (benmarten) [inline AndroidSDK/src/com/leanplum/LeanplumResources.java]: can we please add a comment about why its suppressed?

**Pull request merged**